### DISCUSSION TITLE
Context feature suggestion

### DISCUSSION THREAD

**Discussion by oxidian24** [explosion/spaCy#12217] [New Features & Project Ideas]
I would suggest this thing, I think you call it package - Context. I explained it in a video because I like to visualize things, so using HTML or DOM, you can see the idea of structuring things or coloring of the words classified in the same area...

 https://youtu.be/FEPZL_SoNvI 

Yet you can download the example code for html, as you can see the attribute names and values I used. Description is bellow.

<html>
<html>
<head>

<style>
context[type="bible"],context[type="bible,new_testament"] {
  background-color: yellow;
}
context[area="religion,christianity"],context[area="religion,judaism"] {
  background-color: cyan;
}
context[speaker="Paul"] {
  text-decoration: underline;
}

</style>
</head>

<body>
<context type="bible" age="antiquity">
<pharaoh>Pharaoh's</pharaoh>
<chariot area="military" use="technology">chariots
</chariot>
</context><br>
The <context type="bible,new_testament" age="AD_1_century">
<apostle>apostle</apostle></context>
<context type="bible,new_testament" age="AD_1_century">Paul</context> lived in the first century AD. He had a disciple and friend, <context type="bible,new_testament" age="AD_1_century">Timotheus</context>, whom he took with him on a <context area="religion,christianity">missionary journey</context>. Both characters came out of a mixed marriage, so they were partly <context area="religion,judaism">Jews</context> and Greeks at the same time. We know that <context type="bible,new_testament" age="AD_1_century">Paul</context> was in Rome, as he mentions it in his letter to <context type="bible,new_testament" age="AD_1_century">Timotheus</context> saying:
<context source="bible" speaker="Paul" quote="WEB"> "May the <context  area="religion,christianity" classification="god">Lord</context> grant mercy to the house of Onesiphorus, for he often refreshed me, and was not ashamed of my chain,  but when he was in Rome, he sought me diligently, and found me..."</context>
</context><br>

</body>
</html>

So I would like to see the feature so we can add words or tokens to context. Here I use as an example bible as a source of data. But it can be any book, historical, sci-fi, anything like captions from movie, video, blog, comments.

So the most important attributes I see is type to define the topic what article or text discusses. You can have religion topic and a lot of terms from bible like persons or liturgical stuff like altar and holy things. Also there are nations. 

Another important attributes is area like millitary, use like technology and age or timeline to define the time context. For example T-72s are cold war age but they are uses as tanks today... You want to know if Paul lived in 1st century. Esspecially if you would want to analize the sentences and you see a lot of religion or christain terms or even the date which fits to the bible age/timeline... I mean to the era when apostle Paul lived. I think it may be not so hard if I have a bible as a source for Person names, and nouns or verbs which are very very often in bible. So when one finds them in a bible, spacy could say o predict it is bible term.

**Comment by oxidian24** [comment]
I dont see tHe html so link here https://pastebin.com/EEUmRQ2n

**Comment by polm** [comment]
Thanks for the suggestion. I'm not sure I understand it exactly, but it looks like you should be able to do what you want with some combination of NER, spancat, and entity linking. For example, if you have a span "Paul", you can use entity linking to connect it to the entry for the biblical Paul in an external database, which would tell you when he lived.
